<!DOCTYPE html><html lang="en-gb"><head><script async src="https://www.googletagmanager.com/gtag/js?id=UA-57430133-3"></script><script>window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-57430133-3');</script><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width,initial-scale=1"><title>Software Development - Adrien Leger</title><meta name="description" content="Dev StyleOver&hellip;"><meta name="generator" content="Publii Open-Source CMS for Static Site"><link rel="canonical" href="https://adrienleger.com/software.html"><meta property="og:title" content="Software Development"><meta property="og:image" content="https://adrienleger.com/media/posts/7/network2.jpg"><meta property="og:site_name" content="Adrien Leger"><meta property="og:description" content="Dev StyleOver&hellip;"><meta property="og:url" content="https://adrienleger.com/software.html"><meta property="og:type" content="article"><meta name="twitter:card" content="summary"><meta name="twitter:site" content="@AdrienLeger2"><meta name="twitter:title" content="Software Development"><meta name="twitter:description" content="Dev StyleOver&hellip;"><meta name="twitter:image" content="https://adrienleger.com/media/posts/7/network2.jpg"><link rel="amphtml" href="https://adrienleger.com/amp/software.html"><link rel="alternate" type="application/atom+xml" href="https://adrienleger.com/feed.xml"><link rel="alternate" type="application/json" href="https://adrienleger.com/feed.json"><link rel="shortcut icon" href="https://adrienleger.com/media/website/favicon.ico" type="image/x-icon"><link rel="preconnect" href="https://fonts.gstatic.com/" crossorigin><link href="https://fonts.googleapis.com/css?family=Muli:400,400i,600&amp;subset=latin-ext&amp;display=swap" rel="stylesheet"><style>h1,h2,h3,h4,h5,h6,blockquote,.logo,.navbar__menu,.header__reservation{font-family:'Muli',sans-serif}body,.btn,[type=button],[type=submit],button{font-family:'Muli',sans-serif}</style><link rel="stylesheet" href="https://adrienleger.com/assets/css/style.css?v=4aee07858bebedcbbb80d648f03dbfb2"><script type="application/ld+json">{"@context":"http://schema.org","@type":"Article","mainEntityOfPage":{"@type":"WebPage","@id":"https://adrienleger.com/software.html"},"headline":"Software Development","datePublished":"2020-08-19T23:13","dateModified":"2020-09-29T12:16","image":{"@type":"ImageObject","url":"https://adrienleger.com/media/posts/7/network2.jpg","height":900,"width":1600},"description":"Dev StyleOver&hellip;","author":{"@type":"Person","name":"Adrien Leger"},"publisher":{"@type":"Organization","name":"Adrien Leger"}}</script></head><body><div class="container"><header class="header" id="js-header"><nav class="navbar"><button class="navbar__toggle" aria-label="Menu" aria-haspopup="true" aria-expanded="false"><span class="navbar__toggle-box"><span class="navbar__toggle-inner">Menu</span></span></button><ul id="js-navbar-menu" class="navbar__menu"><li><a href="https://adrienleger.com/about.html" target="_self">About</a></li><li><a href="https://adrienleger.com/expertise.html" target="_self">Expertise</a></li><li class="active"><a href="https://adrienleger.com/software.html" target="_self">Software Dev</a></li><li><a href="https://adrienleger.com/dataviz-galery.html" target="_self">Dataviz</a></li><li><a href="https://adrienleger.com/publications-and-communication.html" target="_self">Publications</a></li><li><a href="https://adrienleger.com/teaching-and-presentations.html" target="_self">Presentations/Posters</a></li><li><a href="https://adrienleger.com/contact.html" target="_self">Contact</a></li></ul></nav><a class="header__logo logo" href="https://adrienleger.com/">Adrien Leger</a></header><div class="main"><div class="wrapper post"><article><figure class="post__featured-image"><img src="https://adrienleger.com/media/posts/7/network2.jpg" srcset="https://adrienleger.com/media/posts/7/responsive/network2-xs.jpg 300w, https://adrienleger.com/media/posts/7/responsive/network2-sm.jpg 600w, https://adrienleger.com/media/posts/7/responsive/network2-md.jpg 900w, https://adrienleger.com/media/posts/7/responsive/network2-xl.jpg 1200w, https://adrienleger.com/media/posts/7/responsive/network2-xxl.jpg 1600w" sizes="(max-width: 1600px) 100vw, 1600px" loading="eager" height="900" width="1600" alt=""></figure><header class="post__header"><h1>Software Development</h1></header><div class="post__inner post__entry"><h3>Dev Style</h3><p>Over the last 7 years I have been actively developing dozens of <strong>scripts, APIs and analytic workflows for bioinformatic analyses</strong>. In particular, I released a number of tools for <strong>gene therapy quality control by NGS</strong> as well as for <strong>nucleotide modification detection</strong> by Oxford Nanopore sequencing.Â </p><p><span class="md-line md-end-block"><span><strong>I â¤ï¸ Python</strong></span> and try to write beautiful and easy-to-maintain code, often using object oriented programming. I follow <strong>good programming practices</strong> including: documenting my code, version control, testing, packaging and continuous integration/deployment. I run pretty much all</span><span class="md-line md-end-block"> my data analyses via <a href="https://jupyter.org/" target="_blank" rel="noopener noreferrer">Jupyter Notebooks</a> to keep a track of everything and to easily share the results with collaborators.</span><span class="md-line md-end-block"> When computing speed is an issue, IÂ  often use <strong>multi-threading</strong> and sometimes <strong><span>Cython/C</span></strong>. I know my way in R, but thoroughly despise it. I use Bash all the time, though I would never consider it for a script longer than a few lines. Let's face it, it is ugly as hell.</span></p><p><span class="md-line md-end-block">I develop on <span><strong>Ubuntu </strong></span><span><strong>Linux</strong></span>Â and my production code runs on a Linux/RedHat <span><strong>high performance computing cluster, </strong>but </span>I do my best to write portable multi-platform packages. I have experience in <span><strong>GPU computing</strong></span> and <strong>containerisation </strong>and I am getting up to speed in cloud computing as well.</span></p><p>Most of my projects are openly released on Github under MIT or GPL licence. Have a look at my <a href="https://github.com/a-slide"><img loading="lazy" style="padding: 5px;" src="https://adrienleger.com/media/posts/7/github.png" sizes="(max-width: 1200px) 100vw, 1200px" srcset="https://adrienleger.com/media/posts/7/responsive/github-xs.png 300w, https://adrienleger.com/media/posts/7/responsive/github-sm.png 480w, https://adrienleger.com/media/posts/7/responsive/github-md.png 749w, https://adrienleger.com/media/posts/7/responsive/github-xl.png 1200w" alt="" width="40" height="40">Github Profile.</a></p><p><a href="https://github.com/a-slide"><img loading="lazy" src="https://github-readme-stats.vercel.app/api?username=a-slide&amp;count_private=true&amp;show_icons=true" data-is-external-image="true" align="left"></a></p><p>Â </p><p>Â </p><p>Â </p><p>Â </p><h3>Top repositories I developed and maintain</h3><p><script src="https://tarptaeya.github.io/repo-card/repo-card.js"><br></script></p><p>Â </p><p><a href="https://a-slide.github.io/pycoQC/"><img loading="lazy" style="padding: 5px;" src="https://adrienleger.com/media/posts/7/pycoQC.svg" alt="pycoQC" width="200rem"></a></p><div class="repo-card" data-repo="a-slide/pycoQC">Â </div><pre>PycoQC started as a lightweight python API for Nanopore data quality control written for Jupyter Notebook. Then I decided to beef it up by experimenting with Plotly to generate nicer interactive plots allowing to explore the data further ğŸ“Š. Along the way it has gained many new functionalities and is now one of the most popular nanopore QC tool, downloaded more than 60,000 times ğŸ˜². </pre><p><a href="https://nanocompore.rna.rocks"><img loading="lazy" style="padding: 5px;" src="https://adrienleger.com/media/posts/7/Nanocompore.svg" alt="Nanocompore" width="400rem"></a></p><div class="repo-card" data-repo="tleonardi/nanocompore">Â </div><pre>NanoCompore is a python command line tools co-developed with Tommaso Leonardi to detect RNA modification by comparing a sample without modifications to an experimental sample. When we started working on NanoCompore we were very disappointed in the existing methods and wanted to write a robust and easy to use alternative. In our hands Nanocompore is one of the best performing option in terms of accuracy.</pre><p><a href="https://a-slide.github.io/pycoMeth/"><img loading="lazy" style="padding: 5px;" src="https://adrienleger.com/media/posts/7/pycoMeth.svg" alt="pycoMeth" width="200rem"></a></p><div class="repo-card" data-repo="a-slide/pycoMeth">Â </div><pre>pycoMeth is a set of tools written in Python to analyse CpG methylation from Nanopore data down to differential methylation analysis. Similar to pycoQC it generates a user friendly interactive HTML report allowing users to explore their data and identify interesting differentialy methylated candidate regions between 2 or more conditions. It is still in active development but has already been download more than 10,000 times.</pre><p><a href="https://a-slide.github.io/NanoCount/"><img loading="lazy" style="padding: 5px;" src="https://adrienleger.com/media/posts/7/NanoCount.svg" alt="NanoCount" width="200rem"></a></p><div class="repo-card" data-repo="a-slide/NanoCount">Â </div><pre>NanoCount is a simple Expectation-Maximisation algorithm to estimate transcript abundance from Nanopore direct-RNA datasets. It is extremely fast and provide accurate estimates.</pre><p><a href="https://a-slide.github.io/NanoSnake/"><img loading="lazy" style="padding: 5px;" src="https://adrienleger.com/media/posts/7/NanoSnake.svg" alt="NanoSnake" width="200rem"></a></p><div class="repo-card" data-repo="a-slide/NanoSnake">Â </div><pre>NanoSnake contains a collection of modular ğŸ Snakemake pipelines to streamline Nanopore and Illumina data analyses, including RNA expression, DNA methylation, Structural variations... The pipelines are ridiculously simple to deploy on local machines, high performance computing clusters or in the Cloud. Conda is the only requirement, everything else is automatically installed when needed.</pre><p>Â </p><div class="repo-card" data-repo="a-slide/pyBioTools">Â </div><pre>Contains small utilities I have been developing over the time to fill-in gaps in existing bioinformatic libraries such as BAM file indexing by read ids or FASTQ files merging and filtering. For a reason I don't really comprehend it has been downloaded more than 8,000 times ğŸ˜•</pre><p>Â </p><div class="repo-card" data-repo="a-slide/Sekator">Â </div><pre>Sekator is a python/C hybrid program to detect and trim off any sequence in any location in sequencing reads. It was an exciting project were I got to interface a high level python interface with a fast C Smith and Waterman Alignment algorithm with a binding layer written in Cython. As a result it is very fast and accurate ğŸš€</pre><p>Â </p></div><footer><div class="post__inner"><div class="post__share"><strong>Share: </strong><a href="https://www.facebook.com/sharer/sharer.php?u=https%3A%2F%2Fadrienleger.com%2Fsoftware.html" class="js-share invert facebook" title="Share with Facebook" rel="nofollow noopener noreferrer">Facebook </a><a href="https://twitter.com/share?url=https%3A%2F%2Fadrienleger.com%2Fsoftware.html&amp;via=%40AdrienLeger2&amp;text=Software%20Development" class="js-share invert twitter" title="Share with Twitter" rel="nofollow noopener noreferrer">Twitter </a><a href="http://www.linkedin.com/shareArticle?url=https%3A%2F%2Fadrienleger.com%2Fsoftware.html&amp;title=Software%20Development" class="js-share invert linkedin" title="Share with LinkedIn" rel="nofollow noopener noreferrer">LinkedIn </a><a href="https://api.whatsapp.com/send?text=Software%20Development https%3A%2F%2Fadrienleger.com%2Fsoftware.html" class="js-share invert whatsapp" title="Share with WhatsApp" rel="nofollow noopener noreferrer">WhatsApp</a></div></div></footer></article></div><footer id="contact" class="section section--dark footer"><div class="footer__overlay"><div><div><p><script async="" src="https://platform.twitter.com/widgets.js" charset="utf-8"></script></p><p><a class="twitter-timeline" data-lang="en" data-width="750" data-dnt="true" data-theme="dark" data-chrome="nofooter noborders noborders transparent noscrollbar" data-tweet-limit="10" href="https://twitter.com/AdrienLeger2?ref_src=twsrc%5Etfw">Tweets by AdrienLeger2</a></p></div><div class="footer__copy">&copy; 2020 <a rel="nofollow">Adrien Leger</a></div><button class="footer__bttop btn--color js-footer__bttop" aria-label="Back to top"><svg><title>Back to top</title><use xlink:href="https://adrienleger.com/assets/svg/svg-map.svg#toparrow"/></svg></button></div></div></footer></div></div><script>window.publiiThemeMenuConfig = {    
           mobileMenuMode: 'overlay',
           animationSpeed: 300,
           submenuWidth: 400,
           doubleClickTime: 500,
           mobileMenuExpandableSubmenus: true, 
           relatedContainerForOverlayMenuSelector: '#js-header',
      };</script><script defer="defer" src="https://adrienleger.com/assets/js/scripts.min.js?v=f4263a053b802221f40fd991e6e1c9cc"></script><script>var images = document.querySelectorAll('img[loading]');
   
         for (var i = 0; i < images.length; i++) {
            if (images[i].complete) {
                  images[i].classList.add('is-loaded');
            } else {
                  images[i].addEventListener('load', function () {
                     this.classList.add('is-loaded');
                  }, false);
            }
         }</script><script defer="defer" src="https://adrienleger.com/assets/js/quicklink.umd.js?v=61daff262ce6be0813e6b1b0341274d4"></script><script>window.addEventListener('load', () =>{
            quicklink();
         });</script></body></html>